src/sql/workbench/services/objectExplorer/browser/filterDialog/filterDialog.ts

```diff
@@ -0,0 +1,678 @@
+						const index = item.filterPropertyIndex;
+						const nodeOperator = this._treeNode.filterProperties[index].type;
+						if (nodeOperator === NodeFilterPropertyDataType.Date || nodeOperator === NodeFilterPropertyDataType.Number) {
+							if (value === BETWEEN_SELECT_BOX || value === NOT_BETWEEN_SELECT_BOX) {
+
+								const tableData = this.filterTable.getData().getItems();
+								if (tableData.length > row + 1) {
+									if (tableData[row + 1].operator.value === AND_SELECT_BOX) {
+										return;
+									}
+								}
+								const newRow: Slick.SlickData = {
+									property: {
+										value: ''
+									},
+									operator: {
+										value: AND_SELECT_BOX,
+										values: [AND_SELECT_BOX]
+									},
+									value: {
+										value: '',
+										values: []
+									},
+									filterPropertyIndex: tableData[row].filterPropertyIndex
+								};
+								const activeElement = this.filterTable.activeCell;
+								tableData.splice(row + 1, 0, newRow);
+								dataProvider.clear();
+								dataProvider.push(tableData);
+								this.filterTable.rerenderGrid();
+								this.filterTable.layout(new DOM.Dimension(600, (dataProvider.getItems().length + 2) * TableRowHeight));
+								this.filterTable.setActiveCell(activeElement.row, activeElement.cell);
+							} else {
+								const tableData = this.filterTable.getData().getItems();
+								if (tableData.length > row + 1) {
+									if (tableData[row + 1].operator.value === AND_SELECT_BOX) {
+										const activeElement = this.filterTable.activeCell;
+										tableData.splice(row + 1, 1);
+										dataProvider.clear();
+										dataProvider.push(tableData);
+										this.filterTable.rerenderGrid();
+										this.filterTable.layout(new DOM.Dimension(600, (dataProvider.getItems().length + 2) * TableRowHeight));
+										this.filterTable.setActiveCell(activeElement.row, activeElement.cell);
+									}
+								}
+							}
+						}
+					}
+				},
+				optionsGetter: (item, column): string[] => {
+					return item[column.field].values;
+				},
+				editorStyler: (component) => {
+					this.styleComponent(component);
+				},
+				onStyleChange: this._onStyleChangeEventEmitter.event
+			}, this._contextViewProvider
+		);
+		const columns: Slick.Column<Slick.SlickData>[] = [
+			{
+				id: PROPERTY_COLUMN_ID,
+				name: PROPERTY_NAME_COLUMN_HEADER,
+				field: PROPERTY_COLUMN_ID,
+				formatter: textFormatter,
+				width: 180,
+			},
+			{
+				id: OPERATOR_COLUMN_ID,
+				name: OPERATOR_COLUMN_HEADER,
+				editor: this._tableCellEditorFactory.getDropdownEditorClass(this, [], false),
+				field: OPERATOR_COLUMN_ID,
+				formatter: textFormatter,
+				width: 180
+			},
+			{
+				id: VALUE_COLUMN_ID,
+				name: VALUE_COLUMN_HEADER,
+				width: 180,
+				formatter: textFormatter,
+				field: VALUE_COLUMN_ID
+			}
+		];
+
+		const clearValueColumn = new ButtonColumn({
+			id: CLEAR_COLUMN_ID,
+			iconCssClass: Codicon.close.classNames,
```

Comment:
What about using an eraser icon instead? The x to me more means "delete this row"

Reply:
agreed, I thought I would remove the whole filter clause, but then realized we don't have an Add button. definitely should use erase icon.

Reply:
@Charles-Gagnon  and @alanrenmsft  added erase icon.
![image](https://user-images.githubusercontent.com/6816294/236575270-c8e5bac6-6bb2-47d3-8874-a0da14079eb7.png)